wgengine/filter/filter.go

```diff
@@ -184,23 +185,27 @@ func New(matches []Match, localNets, logIPs *netipx.IPSet, shareStateWith *Filte
 		}
 	}
 
-	containsFunc := func(s *netipx.IPSet) func(netip.Addr) bool {
-		if s == nil {
-			return tsaddr.FalseContainsIPFunc()
-		}
-		return tsaddr.NewContainsIPFunc(views.SliceOf(s.Prefixes()))
-	}
-
 	f := &Filter{
 		logf:     logf,
 		matches4: matchesFamily(matches, netip.Addr.Is4),
 		matches6: matchesFamily(matches, netip.Addr.Is6),
 		cap4:     capMatchesFunc(matches, netip.Addr.Is4),
 		cap6:     capMatchesFunc(matches, netip.Addr.Is6),
-		local:    containsFunc(localNets),
-		logIPs:   containsFunc(logIPs),
+		local4:   tsaddr.FalseContainsIPFunc(),
+		local6:   tsaddr.FalseContainsIPFunc(),
+		logIPs:   tsaddr.FalseContainsIPFunc(),
 		state:    state,
 	}
+	if localNets != nil {
+		p := localNets.Prefixes()
+		p4, p6 := slicesx.Partition(p, func(p netip.Prefix) bool { return p.Addr().Is4() })
+		f.local4 = tsaddr.NewContainsIPFunc(views.SliceOf(p4))
+		f.local6 = tsaddr.NewContainsIPFunc(views.SliceOf(p6))
+	}
+	if logIPs != nil {
```

Comment:
why not the same for `logIPs`?

Reply:
Haven't gotten there yet. It's more invasive because the callers are a mix of v4 and v6.

Reply:
done